site/content/blog/2020-09-03-zap-jwt-scanner/index.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "ZAP JWT Support Add-on"
+description: "How to scan and fuzz JWT's using OWASP ZAP"
+images:
+- https://www.zaproxy.org/blog/2020-09-03-zap-jwt-scanner/images/jwt-options-panel.png
+type: post
+tags:
+- blog
+date: "2020-09-03"
+addSocialPreview: true
+authors:
+    - preetkaran20
+---
+
+With the popularity of JSON Web Tokens (JWTs) there comes the need to secure their use so that they are not misused because of bad configuration, older libraries, or buggy implementations. So the JWT Support add-on is used to find such vulnerabilities and this blog explains on how to use it.
+
+## Configuration
+As JWT add-on includes a rule for the Active Scanner and Fuzzer but there are configuration details which are specific for the JWT add-on.
+Under ZAP's Options dialog you will find a JWT section as shown below:
+![JWT](./images/jwt-options-panel.png)
+ 
+### Explanation
+
+#### Scanner Configuration: 
+
+In case the application which you are trying to scan is using RSA or more specifically RS* algorithm then please configure the public certificate TrustStore path and TrustStore password. These fields are used to find certain vulnerabilities related to RS* based JWTs.
+
+Enable Client Configuration Scan flag is used to enable client side validations like JWT being sent to the browser in an insecure or non-recommended way.
+
+#### Fuzzer Configuration:
+
+As JWT is a signed token hence fuzzing field values require resigning the JWT and hence for signing fuzzer requires HMac secret key or RSA's private key as per the algorithm header field of JWT. So Fuzzer configurations corresponds to the same.
```

Comment:
```suggestion
Since JWT is a signed token; fuzzing field values requires resigning the JWT therefore the fuzzer requires an HMac secret key or RSA private key as per the algorithm header field of the JWT. So that the Fuzzer configuration corresponds to the same.
```